### PR TITLE
feat(plugin_api): Align DispatchOpFn signature to that of OpFn and access to OpState

### PR DESCRIPTION
**Pull request by eliassjogreen**
This pr adds plugin access to the deno resource table through a `get_resource_table` method in the `plugin_api::Interface`. This will be really useful for [webgpu windowing and swapchains](https://github.com/denoland/deno/issues/7863#issuecomment-706897139) but also just for plugins in general not having to use their own resource table solution (usually a global HashMap) for storing resources.

### CODE REVIEW

**Comment by lucacasonato** [comment]
I'm not really super happy with this. I would much prefer if we finally aligned op crates and plugins, and have them have the same signature (the one stating with `deno_core::OpState`).

**Comment by lucacasonato** [comment]
Can you also add a test checks if `deno_core::json_op_async` and `deno_core::json_op_sync` work now?

**Comment by eliassjogreen** [comment]
Sure, it seems to work fine when using both `json_op_sync` and `json_op_async` and the internal `core.jsonOpSync` and `core.jsonOpAsync` functions for calling and getting a response.

File: test_plugin/src/lib.rs [hunk 1]
```diff
@@ -88,3 +102,27 @@ fn op_test_resource_table_close(
...
+  Ok(json!(format!("Hello {}!", args.name)))
+}
+
+pub async fn op_test_json_op_async(
+  _state: Rc<RefCell<OpState>>,
+  args: JsonOpArgs,
+  _bufs: BufVec,
+) -> Result<Value, AnyError> {
+  println!("Hello {} from async json op!", args.name);
+  Ok(json!(format!("Hello {}!", args.name)))
```
Comment (littledivy) [inline test_plugin/src/lib.rs:127]: Would make sense to do some async tasks inside this op?

Comment (eliassjogreen) [reply test_plugin/src/lib.rs]: Makes a lot of sense :P

**Pull request closed without merging**